test/Bugs/dataview_setfloat_nan_if_undefined.js

```diff
@@ -0,0 +1,24 @@
+var buffer = new ArrayBuffer(8);
+var sample = new DataView(buffer, 0);
+var passCnt = 0;
+try {
+  sample.setFloat32(0);
+  if (isNaN(sample.getFloat32(0))) passCnt++;
+  sample.setFloat64(0);
+  if (isNaN(sample.getFloat64(0))) passCnt++;
+  sample.setInt32(0);
+  if (sample.getInt32(0) === 0) passCnt++;
+  sample.setInt16(0);
+  if (sample.getInt16(0) === 0) passCnt++;
+  sample.setInt8(0);
+  if (sample.getInt8(0) === 0) passCnt++;
+  sample.setUint32(0);
+  if (sample.getUint32(0) === 0) passCnt++;
+  sample.setUint16(0);
+  if (sample.getUint16(0) === 0) passCnt++;
+  sample.setUint8(0);
+  if (sample.getUint8(0) === 0) passCnt++;
+} catch (err) { }
+
+if (passCnt === 8) print("PASSED")
+else print("FAILED")
```

Comment:
This looks like a good test, though it would be good to avoid adding a new file for it if we can - at some point I'd like us to combine a lot of the old test files, this could be slotted into the existing test/typdearray/dataview2.js test file.

Reply:
If you move the tests into that file you can use use the `assert` methods, which can help with better error reporting.

Reply:
Done.